src/codeu/chat/client/ClientConversation.java

```diff
@@ -51,21 +51,39 @@ public ClientConversation(Controller controller, View view, ClientUser userConte
     this.userContext = userContext;
   }
 
+  //Rename the title of the conversation
+  public void renameConversation(String newTitle) {
+    if (isValidTitle(newTitle)) {
+      controller.renameConversation(currentConversation.id, newTitle);
+    } else {
+      LOG.error("Error: new name is invalid.");
+    }
+  }
+
   public void setMessageContext(ClientMessage messageContext) {
     this.messageContext = messageContext;
   }
 
   // Validate the title of the conversation
-  static public boolean isValidTitle(String title) {
+  public boolean isValidTitle(String title) {
     boolean clean = true;
     if ((title.length() <= 0) || (title.length() > 64)) {
       clean = false;
+    } else if(titleExists(title)) {
+      clean = false;
     } else {
+      // Other conditions go here
```

Comment:
what other conditions??? XD

make a todo or remove

Reply:
Mayhaps "clean", no-swear-word titles is what I'm considering.